ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +50,142 @@ const TRANSIT_PARAMS = {
   rewrap: ['ciphertext', 'context', 'nonce', 'key_version'],
+  datakey: [],
 };
-const SUCCESS_MESSAGE_FOR_ACTION = {
-  sign: 'Signed your data',
-  // the verify action doesn't trigger a success message
-  hmac: 'Created your hash output',
-  encrypt: 'Created a wrapped token for your data',
-  decrypt: 'Decrypted the data from your token',
-  rewrap: 'Created a new token for your data',
-  datakey: 'Generated your key',
-  export: 'Exported your key',
-};
-export default Component.extend(TRANSIT_PARAMS, {
-  store: service(),
-  flashMessages: service(),
-
-  // public attrs
-  selectedAction: null,
-  key: null,
-  isModalActive: false,
-
-  onRefresh() {},
-  init() {
-    this._super(...arguments);
-    // TODO figure out why get is needed here Ember Upgrade
-    // eslint-disable-next-line ember/no-get
-    if (this.selectedAction) {
-      return;
-    }
-    // eslint-disable-next-line ember/no-get
-    set(this, 'selectedAction', get(this, 'key.supportedActions.firstObject'));
-    assert('`key` is required for `' + this.toString() + '`.', this.getModelInfo());
-  },
-
-  didReceiveAttrs() {
-    this._super(...arguments);
-    this.checkAction();
-    if (this.selectedAction === 'export') {
-      this.setExportKeyDefaults();
-    }
-  },
-
-  setExportKeyDefaults() {
-    const exportKeyType = this.key.exportKeyTypes.firstObject;
-    const exportKeyVersion = this.key.validKeyVersions.lastObject;
-    this.setProperties({
-      exportKeyType,
-      exportKeyVersion,
-    });
-  },
-
-  keyIsRSA: computed('key.type', function () {
-    const type = this.key.type;
-    return type === 'rsa-2048' || type === 'rsa-3072' || type === 'rsa-4096';
-  }),
 
-  getModelInfo() {
-    const model = this.key || this.backend;
-    if (!model) {
-      return null;
-    }
-    const backend = model.backend || model.id;
-    const id = model.id;
-
-    return {
-      backend,
-      id,
-    };
-  },
-
-  checkAction() {
-    const currentAction = this.selectedAction;
-    const oldAction = this.oldSelectedAction;
-
-    this.resetParams(oldAction, currentAction);
-    set(this, 'oldSelectedAction', currentAction);
-  },
-
-  resetParams(oldAction, action) {
-    const params = copy(TRANSIT_PARAMS);
-    let paramsToKeep;
-    const clearWithoutCheck =
-      !oldAction ||
-      // don't save values from datakey
-      oldAction === 'datakey' ||
-      // can rewrap signatures — using that as a ciphertext later would be problematic
-      (oldAction === 'rewrap' && !this.key.supportsEncryption);
-
-    if (!clearWithoutCheck && action) {
-      paramsToKeep = PARAMS_FOR_ACTION[action];
+export default class TransitKeyActions extends Component {
+  @service store;
+  @service flashMessages;
+  @service router;
+
+  @tracked isModalActive = false;
+  @tracked errors = null;
+  @tracked props = Object.assign({}, STARTING_TRANSIT_PROPS); // shallow copy of the object. We don't want to mutate the original.
```

Comment:
copy and assign are to be deprecated in ember 5.x.x. The docs indicate we should use native javascript methods instead.

Reply:
👇 these guys.
```
import { assign } from '@ember/polyfills';
import { copy } from 'ember-copy';
```